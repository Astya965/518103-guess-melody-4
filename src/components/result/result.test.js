import React from "react";
import renderer from "react-test-renderer";
import Result from "./result.jsx";

describe(`Render Result`, () => {
  it(`With isSuccess is true`, () => {
    const tree = renderer
      .create(<Result
        isSuccess={true}
        mistakes={2}
        right={5}
        onResetButtonClick={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With isSuccess is false`, () => {
    const tree = renderer
      .create(<Result
        isSuccess={false}
        onResetButtonClick={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
