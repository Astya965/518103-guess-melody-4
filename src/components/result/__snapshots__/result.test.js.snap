// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Result With isSuccess is false 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Какая жалость!
  </h2>
  <p
    className="result__total result__total--fail"
  >
    У вас закончились все попытки. Ничего, повезёт в следующий раз!
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Попробовать ещё раз
  </button>
</section>
`;

exports[`Render Result With isSuccess is true 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total "
  >
    Вы ответили правильно на 5 вопросов и совершили 2 ошибки
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Попробовать ещё раз
  </button>
</section>
`;
